File: demos/www/cgi-bin/delete.php
<?php

// Specify the directory where files are stored
$filesDirectory = '../public/';

// Check if the form was submitted
if ($_SERVER['REQUEST_METHOD'] === 'POST') {
    // Check if a file was selected for deletion
    if (isset($_POST['fileToDelete'])) {
        $fileToDelete = $_POST['fileToDelete'];

        // Construct the full path to the file
        $filePath = $filesDirectory . $fileToDelete;

        // Check if the file exists before attempting to delete
        if (file_exists($filePath)) {
            // Attempt to delete the file
            if (unlink($filePath)) {
                echo <<<HTML
                <html lang="en">
                    <head>
                        <link rel="stylesheet" href="/styles.css">
                        <title>File Upload Form</title>
                    </head>
                    <body>
                        <div class="wrapper">
                            <div class="flex-col">
                                <a href="/index.html">
                                    <img src="/icons/back.svg" alt="" />
                                    Home
                                </a>
                                <h2>File was deleted</h2>
            HTML;
                                echo "<span style='color:red;'>";echo $fileToDelete;echo "</span>";
                                echo <<<HTML
                                <a href="/public">Uploaded files</a>
                                <span style="color:gray;">Powered by PHP/8.2.12</span>
                            </div>
                        </div>
                    </body>
                </html>
                HTML;
            } else {
                echo "Error deleting file: $fileToDelete";
            }
        } else {
            echo "File not found: $fileToDelete";
        }
    } else {
        echo "Please select a file to delete.";
    }
} else {
    // Display the HTML form for selecting and deleting a file
    $files = scandir($filesDirectory);
    $action = htmlspecialchars($_GET['action']);

    echo <<<HTML
<html lang="en">
<head>
    <link rel="stylesheet" href="/styles.css">
    <title>File Upload Form</title>
</head>
<body>
    <div class="wrapper">
        <form action=$action method="post">
            <div class="flex-col">
                <a href="/">
                    <img src="/icons/back.svg" alt="" />
                    Home
                </a>
                <h1 for="file">Select a file to delete</h1>
                <select id="fileToDelete" name="fileToDelete" required>
HTML;

                foreach ($files as $file) {
                    if ($file != '.' && $file != '..') {
                        echo "<option value=\"$file\">$file</option>";
                    }
                }
                echo <<<HTML
                    </select>
                    <button type="submit">Delete</button>
            </div>
        </form>
    </div>
</body>
</html>
HTML;
}

?>

File: demos/www/cgi-bin/delete.py
import cgi
import os

# Specify the directory where files are stored
files_directory = "../public/"

# Print the Content-type header
print("Content-type: text/html\n")

# Parse query string parameters
form = cgi.FieldStorage()
file_to_delete = form.getvalue("fileToDelete")

# Check if a filename was provided
if file_to_delete:
    # Construct the full path to the file
    file_path = os.path.join(files_directory, file_to_delete)

    # Check if the file exists before attempting to delete
    if os.path.exists(file_path):
        # Attempt to delete the file
        try:
            os.remove(file_path)
            print(print(f"""
                <!DOCTYPE html>
                <html lang="en">
                <head>
                    <meta charset="UTF-8">
                    <title>Greeting</title>
                    <link rel="stylesheet" href="/styles.css" />
                </head>
                <body>
                    <div class="wrapper">
                    <div class="flex-col">
                        <a href="/index.html">
                        <img src="/icons/back.svg" alt="" />
                        Home</a
                        >
                        <h2>File deleted successfully</h2>
                        <span style='color:red;'>{file_to_delete}</span>
                        <a href="/public">Uploaded files</a>
                        <span style="color:gray;">Powered by Python</span>
                    </div>
                    </div>
                </body>
                </html>
                """))
        except Exception as e:
            print(f"Error deleting file: {e}")
    else:
        print(f"File not found: {file_to_delete}")
else:
    print("Please provide a filename to delete.")